pkg/apis/work/v1alpha1/binding_types.go

```diff
@@ -54,12 +58,23 @@ type ObjectReference struct {
 	// determine when object has changed.
 	// +optional
 	ResourceVersion string `json:"resourceVersion,omitempty"`
+
+	// ResourcePerReplicas represents the resource each replica needs
+	// +optional
+	ResourcePerReplicas corev1.ResourceList `json:"resourcePerReplicas,omitempty"`
+
+	// Replicas represents the replicas of the object
+	// +optional
+	Replicas int64 `json:"replicas,omitempty"`
```

Comment:
The two fields seem used to `record` something, I don't think it's a good idea to add them to the `ObjectReference`, because this is used for `referencing` the resource template.

Not sure if we really need them from the limited comments. If yes, they should be in the `Status`.

Reply:
The two fields are used during the scheduler. We need to know this info so that we can assign the replicas and schedule according the resource.

Reply:
Got it.